### PR TITLE
Update to Apache Beam, v2.1.0.

### PR DESCRIPTION
**Pull request by wcn3**
Updated the compiler plugin version, and added explicit versioning for the JAR plugin.

### CODE REVIEW

**Comment by okorz001** [comment]
Did you mean to update `beam.version`?

**Comment by wcn3** [comment]
That appears to be the case. That was not listed as a step in the instructions I was following. There is also an obvious deficiency in that there are no source files associated in this change. I need to address both of these problems.

**Comment by wcn3** [comment]
This PR has been updated and should contain all the necessary changes.

**Comment by lukecwik** [comment]
Travis failed in testing the archetype:
```
[INFO] [ERROR] /Users/travis/build/GoogleCloudPlatform/DataflowJavaSDK/maven-archetypes/examples/target/test-classes/projects/basic/project/basic/src/main/java/it/pkg/common/WriteOneFilePerWindow.java:[25,44] cannot find symbol
[INFO]   symbol:   class OutputFileHints
[INFO]   location: class org.apache.beam.sdk.io.FileBasedSink
[INFO] [ERROR] /Users/travis/build/GoogleCloudPlatform/DataflowJavaSDK/maven-archetypes/examples/target/test-classes/projects/basic/project/basic/src/main/java/it/pkg/common/WriteOneFilePerWindow.java:[97,40] cannot find symbol
[INFO]   symbol:   class OutputFileHints
[INFO]   location: class it.pkg.common.WriteOneFilePerWindow.PerWindowFiles
[INFO] [ERROR] /Users/travis/build/GoogleCloudPlatform/DataflowJavaSDK/maven-archetypes/examples/target/test-classes/projects/basic/project/basic/src/main/java/it/pkg/common/WriteOneFilePerWindow.java:[113,41] cannot find symbol
[INFO]   symbol:   class OutputFileHints
[INFO]   location: class it.pkg.common.WriteOneFilePerWindow.PerWindowFiles
```

File: maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -79,6 +79,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
```
Comment (lukecwik) [inline maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml]: exec-maven-plugin should be version 1.6.0

**Pull request merged**